paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -771,26 +773,42 @@ void MultiDevSSAGraphBuilder::CreateRPCOp(ir::Graph *result,
                    "This hack no longer holds, please fix.");
     // the variable name which contains .block means it was splited by
     // split_byref op
-    // so that we can balance the variable blocks to all the pserver
-    // instances.
     if (strategy_.reduce_ == BuildStrategy::ReduceStrategy::kAllReduce &&
         node->inputs[0]->Name().find(".block") == std::string::npos) {
       std::vector<std::string> input_var_names;
       for (ir::Node *n : node->inputs) {
         input_var_names.push_back(n->Name());
       }
```

Comment:
I suspect input_var_names is no longer needed.

Reply:
It's still needed to set the input actual place, input variable may not be the same as the op_role_var, it may be splited into several vars.

Reply:
And https://github.com/PaddlePaddle/Paddle/pull/12899 is polishing the deps build.